src/main/java/util/TickingTimerManager.java

```diff
@@ -0,0 +1,95 @@
+package util;
+
+/**
+ * This class manages the TickingTimer instance used for refreshing. It computes the refresh rate period and change
+ * the refresh period.
+ */
+public class TickingTimerManager {
+
+    /**
+     * To allow some additional time until next refresh.
+     */
+    public static final int BUFFER_TIME = 1;
+
+    /**
+     * The amount that is set aside for manual refresh or creation of issues, labels and milestones.
+     */
+    public static final int APIQUOTA_BUFFER = 200;
+
+    public static final int NO_OF_REFRESH_THAT_REQUIRES_MORE_TIME = 1;
+    public static final int DEFAULT_REFRESH_PERIOD_IN_SEC = 60;
+
+    private final TickingTimer timer;
+
+    public TickingTimerManager(TickingTimer timer) {
+        this.timer = timer;
+    }
+
+    public void startTimer() {
+        timer.start();
+    }
+
+    public void restartTimer() {
+        timer.restart();
+    }
+
+    /**
+     * Computes the TickerTimer period that are used for refreshing the issues periodically.
+     * @param apiQuota         : The remaining allowed api request until the next api request allowance renewal.
+     * @param remainingTime    : The remaining time left until the next api request allowance renewal.
+     * @param lastApiCallsUsed : The amount of api used in the last api pull.
+     */
+    public long computeTickerTimerPeriod(int apiQuota, long remainingTime, int lastApiCallsUsed) {
+
+        assert apiQuota >= 0 && remainingTime >= 0 && lastApiCallsUsed >= 0;
+
+        long refreshTimeInMin;
+
+        if (isDuringAppInitOrRemainingTimeIsZero(apiQuota, remainingTime, lastApiCallsUsed)) {
```

Comment:
Is it more readable to use a variable here instead of a separate method? e.g.
```java
boolean isDuringAppInit = apiQuota != 0 && lastApiCallsUsed == 0;
if (isDuringAppInit || remainingTime == 0) {
...
}